File: Python/divide_andConquer.py



a = ['1', '5', '8', 8, 9, 9, 2, '3'] #11




somaInt = 0
somaStr = 0
for x in a:
    if type(x)==int:
        somaInt+=x
    else:
        somaStr+=int(x)

print(somaInt-somaStr)

File: Rust/divide_and_conquer.rs
use either::Either;

fn div_con(arr: &[Either<i32, String>]) -> i32 {
    let mut somaInt:i32 = 0;
    let mut somaStr:i32 = 0;
    for x in arr{
        match x{
            Either::Left(x)=> somaInt+=x,
            Either::Right(x)=> somaStr+=x.parse::<i32>().unwrap(),
            
        }
    }
    somaInt - somaStr
}
